artichoke-backend/src/extn/core/time/trampoline.rs

```diff
@@ -472,10 +467,30 @@ pub fn subsec(interp: &mut Artichoke, time: Value) -> Result<Value, Error> {
 
 // Time format
 
-pub fn strftime(interp: &mut Artichoke, time: Value, format: Value) -> Result<Value, Error> {
-    let _ = interp;
-    let _ = time;
-    let _ = format;
-    // Requires a parser.
-    Err(NotImplementedError::new().into())
+pub fn strftime(interp: &mut Artichoke, mut time: Value, format: Value) -> Result<Value, Error> {
+    let time = unsafe { Time::unbox_from_value(&mut time, interp)? };
+    let format = interp.try_convert_mut(format)?;
+
+    let bytes: Vec<u8> = time.strftime(format).map_err(|e| {
+        // InvalidFormatString is the only true ArgumentError, where as the
+        // rest that can be thrown from strftime are runtime failures.
+        //
+        // ```console
+        // [3.1.2]> Time.now.strftime("%")
+        // (irb):1:in `strftime': invalid format: % (ArgumentError)
+        //      from (irb):1:in `<main>'
+        //	    from /home/ben/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
+        //	    from /home/ben/.rbenv/versions/3.1.2/bin/irb:25:in `load'
+        //	    from /home/ben/.rbenv/versions/3.1.2/bin/irb:25:in `<main>'
+        // ```
+        if let strftime::Error::InvalidFormatString = e {
```

Comment:
Can we restructure this as an "exhaustive" (I think the error enum is marked non exhaustive) match? I'd feel more comfortable with this squash into runtime error approach if we are explicitly enumerating the cases we are squashing (at the very least it means we don't have to open the docs to see).

Reply:
Yes good point, it would make it a lot more explicit (good thing). I'm imagine a match block with each enum value, is that what you had in mind too?

Reply:
yup that's what I had in mind. 👍

Reply:
Some limitations with regards to the non-exhaustive list, however I've added a comment to cater for this. I could have ignored the clippy error instead, but will resolve this for now.